lib/spack/spack/binary_distribution.py

```diff
@@ -105,6 +106,9 @@ def __init__(self, cache_root):
         # cache (_mirrors_for_spec)
         self._specs_already_associated = set()
 
+        # mapping from mirror urls to the time.time() of the last index fetch.
+        self._last_fetch_times = {}
```

Comment:
And as far as I can see: this does not help across two `spack install` calls :(

Reply:
There was concern for sequences of operations like:
```
$ spack install ...
$ spack buildcache update-index ...
$ spack install ...
```
If the fetch timers spanned `spack install` commands, the latter `spack install` may (surprisingly) not see the effect of the `spack buildcache update-index`. Same logic as to why `update()` has a `with_cooldown=False` argument.

Note that `index.json.hash` gets fetched first before `index.json` (or at least, it's supposed to). So if the `index.json` didn't change between calls it won't be fetched a second time.

Reply:
We otherwise invalidate cache by mtime.